Humbertino_Avilez/CRUD-EXCEPTION/Vet_Patitas/src/main/java/com/bolsadeideas/springboot/bootcamp/app/service/implement/AnimalServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.bolsadeideas.springboot.bootcamp.app.service.implement;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.bolsadeideas.springboot.bootcamp.app.dao.AnimalDao;
+import com.bolsadeideas.springboot.bootcamp.app.entity.Animal;
+import com.bolsadeideas.springboot.bootcamp.app.exception.AnimalNotFoundException;
+import com.bolsadeideas.springboot.bootcamp.app.service.InterfaceAnimalService;
+import static com.bolsadeideas.springboot.bootcamp.app.utils.Constants.NO_FOUND_ANIMAL;
+
+@Service
+public class AnimalServiceImpl implements InterfaceAnimalService {
+
+	@Autowired
+	private AnimalDao animalDao;
+
+	@Override
+	@Transactional(readOnly = true)
```

Comment:
@Transactional???
do you need findAll() and findById() being part of a Transaction??
don't get it..... would you explain to me please?

Reply:
In my defense  I was following a course, it mention that I should write it because it has best performance cause the stats of the objects are not saved and they are read-only

Reply:
Interesting... but, not buying it jajajaja